client/src/context/UserContext.ts

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+export interface UserType {
+    username: string;
+}
+
+export interface User {
```

Comment:
Would it be better to name this `UserState`, and replace `UserType` with `User`?